bikeshed/dfnpanels.py

```diff
@@ -45,7 +45,10 @@ def addDfnPanels(doc, dfns):
             for i, el in enumerate(els):
                 refID = el.get("id")
                 if refID is None:
-                    refID = f"ref-for-{id}"
+                    if hasAncestor(el, lambda x: hasClass(x, "domintro")):
```

Comment:
I'd rather abstract this to a method in h.dom, just to give it a bit more semantic meaning.

Reply:
Actually, since this full conditional block is duplicated in unsortedtext, it would probably be good to shift the entire thing over, to a `generateRefId` method or something. (Feel free to use a better name.)

Reply:
> Actually, since this full conditional block is duplicated in unsortedtext, it would probably be good to shift the entire thing over, to a `generateRefId` method or something. (Feel free to use a better name.)

OK will do — and to be clear, this `generateRefId` method should best be in h.dom?

Reply:
> I'd rather abstract this to a method in h.dom, just to give it a bit more semantic meaning.

OK, done now